docs/data/date-pickers/date-calendar/ReverseYearCalendar.tsx

```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import dayjs from 'dayjs';
+import { DemoContainer, DemoItem } from '@mui/x-date-pickers/internals/demo';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { YearCalendar } from '@mui/x-date-pickers/YearCalendar';
+
+export default function ReverseYearCalendar() {
+  const currentYear = dayjs();
```

Comment:
In most if not all cases static values are defined outside of the component, WDYT about moving it before the component function? 🤔